modules/coreos-layering-configuring-on-modifying.adoc

```diff
@@ -0,0 +1,139 @@
+// Module included in the following assemblies:
+//
+// * machine_configuration/coreos-layering.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="coreos-layering-configuring-on-modifying_{context}"]
+= Modifying a custom layered image
+
+You can modify an on-cluster custom layered image, as needed. This allows you to install additional packages, remove existing packages, change the pull or push repositories, update secrets, or other similar changes. You can edit the `MachineOSConfig` object, apply changes to the YAML file that created the `MachineOSConfig` object, or create a new YAML file for that purpose.
+
+If you modify and apply the `MachineOSConfig` object YAML or create a new YAML file, the YAML overwrites any changes you made directly to the `MachineOSConfig` object itself.
+
+include::snippets//coreos-layering-configuring-on-pause.adoc[]
+
+.Prerequisites
+
+* You have opted in to on-cluster layering by creating a `MachineOSConfig` object.
+
+.Procedure
+
+* Modify an object to update the associated custom layered image:
+
+.. Edit the `MachineOSConfig` object to modify the custom layered image. The following example adds the `rngd` daemon to nodes that already have the tree package that was installed using a custom layered image.
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1alpha1
+kind: MachineOSConfig
+metadata:
+  name: layered-alpha1
+spec:
+  machineConfigPool:
+    name: layered
+  buildInputs:
+    containerFile:
+    - containerfileArch: noarch
+      content: |- <1>
+        FROM configs AS final
+
+        RUN rpm-ostree install rng-tools && \
+            systemctl enable rngd && \
+            rpm-ostree cleanup -m && \
+            ostree container commit
+
+        RUN rpm-ostree install tree && \
+            ostree container commit
+    imageBuilder:
+      imageBuilderType: PodImageBuilder
+    baseImagePullSecret:
+      name: global-pull-secret-copy <2>
+    renderedImagePushspec: image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-images:latest <3>
+    renderedImagePushSecret:  <4>
+      name: new-secret-name
+  buildOutputs:
+    currentImagePullSecret:
+      name: new-secret-name <5>
+----
+<1> Optional: Modify the Containerfile, for example to add or remove packages.
+<2> Optional: Update the secret needed to pull the base operating system image from the registry.
+<3> Optional: Modify the image registry to push the newly-built custom layered image to.
+<4> Optional: Update the secret needed to push the newly-built custom layered image to the registry.
+<5> Optional: Update the secret needed to pull the newly-built custom layered image from the registry.
++
+When you save the changes, the MCO drains, cordons, and reboots the nodes. After the reboot, the node uses the cluster base {op-system-first} image. If your changes modify a secret only, no new build is triggered and no reboot is performed.
```

Comment:
thought: Unless I missed it, we should probably include adding a node to the new MachineConfigPool.